File: spec/code_samples/Python/v1@messaging@numbers@dedicated@assignments/get.py
import base64
import json
import urllib.request

api_key = 'YOUR_API_KEY'
api_secret = 'YOUR_API_SECRET'
api_host = 'YOUR_API_HOST'  # Set YOUR_API_HOST to the regional host from the servers section in the docs
page_size = 0
token = "YOUR_TOKEN"
number_id = "YOUR_NUMBER_ID"
matching = "YOUR_MATCHING"
country = "YOUR_COUNTRY"
type = "YOUR_TYPE"
types = "YOUR_TYPES"
classification = "YOUR_CLASSIFICATION"
service_types = "YOUR_SERVICE_TYPES"
label = "YOUR_LABEL"
sort_by = "TIMESTAMP"
sort_direction = "ASCENDING"

# HMAC authentication is also supported instead of Basic
auth = base64.b64encode(f'{api_key}:{api_secret}'.encode()).decode()
url = f'{api_host}/v1/messaging/numbers/dedicated/assignments?page_size={page_size}&token={token}&number_id={number_id}&matching={matching}&country={country}&type={type}&types={types}&classification={classification}&service_types={service_types}&label={label}&sort_by={sort_by}&sort_direction={sort_direction}'
headers = {
    'Authorization': f'Basic {auth}',
    'Accept': 'application/json',
}

request = urllib.request.Request(url, headers=headers, method='GET')

with urllib.request.urlopen(request) as response:
    print(response.status)
    print(response.read().decode())

File: spec/code_samples/JavaScript/v1@messaging@numbers@dedicated@assignments/get.js
const apiKey = 'YOUR_API_KEY';
const apiSecret = 'YOUR_API_SECRET';
const apiHost = 'YOUR_API_HOST'; // Set YOUR_API_HOST to the regional host from the servers section in the docs
const pageSize = 0;
const token = 'YOUR_TOKEN';
const numberId = 'YOUR_NUMBER_ID';
const matching = 'YOUR_MATCHING';
const country = 'YOUR_COUNTRY';
const type = 'YOUR_TYPE';
const types = 'YOUR_TYPES';
const classification = 'YOUR_CLASSIFICATION';
const serviceTypes = 'YOUR_SERVICE_TYPES';
const label = 'YOUR_LABEL';
const sortBy = 'TIMESTAMP';
const sortDirection = 'ASCENDING';

// HMAC authentication is also supported instead of Basic
const auth = Buffer.from(`${apiKey}:${apiSecret}`).toString('base64');

const response = await fetch(`${apiHost}/v1/messaging/numbers/dedicated/assignments?page_size=${pageSize}&token=${token}&number_id=${numberId}&matching=${matching}&country=${country}&type=${type}&types=${types}&classification=${classification}&service_types=${serviceTypes}&label=${label}&sort_by=${sortBy}&sort_direction=${sortDirection}`, {
  method: 'GET',
  headers: {
    Authorization: `Basic ${auth}`,
    Accept: 'application/json',
  },
});

console.log(response.status);
console.log(await response.text());
